frame/assets/src/lib.rs

```diff
@@ -805,19 +993,19 @@ pub mod pallet {
 		where
 			<T as Config>::MultiCurrency: Inspect<T::AccountId, Balance = T::Balance, AssetId = T::AssetId>
 				+ InspectHold<T::AccountId>,
-			<T as Config>::Currency:
+			<T as Config>::NativeCurrency:
 				NativeInspect<T::AccountId, Balance = T::Balance> + NativeInspectHold<T::AccountId>,
 		{
 			fn balance_on_hold(asset: Self::AssetId, who: &T::AccountId) -> Self::Balance {
 				if asset == T::NativeAssetId::get() {
-					return <<T as Config>::Currency>::balance_on_hold(who)
+					return <<T as Config>::NativeCurrency>::balance_on_hold(who)
 				}
 				<<T as Config>::MultiCurrency>::balance_on_hold(asset, who)
 			}
 
 			fn can_hold(asset: Self::AssetId, who: &T::AccountId, amount: Self::Balance) -> bool {
 				if asset == T::NativeAssetId::get() {
-					return <<T as Config>::Currency>::can_hold(who, amount)
+					return <<T as Config>::NativeCurrency>::can_hold(who, amount)
```

Comment:
not sure about merge strategy here. are you goint to  merge previous PR before this?